spec/rubocop/directive_comment_spec.rb

```diff
@@ -149,25 +149,31 @@
   end
 
   describe '#cop_names' do
-    subject(:cop_names) { directive_comment.cop_names }
+    subject { directive_comment.cop_names }
 
-    context 'when all cops mentioned' do
-      let(:comment_cop_names) { 'all' }
-      let(:global) { instance_double(RuboCop::Cop::Registry, names: names) }
-      let(:names) { %w[all_names Lint/RedundantCopDisableDirective] }
-
-      before { allow(RuboCop::Cop::Registry).to receive(:global).and_return(global) }
-
-      it 'returns all cop names except redundant' do
-        expect(cop_names).to eq(%w[all_names])
-      end
-    end
-
-    context 'when cop specified' do
-      let(:comment_cop_names) { 'Foo/Bar' }
-
-      it 'returns parsed cop names' do
-        expect(cop_names).to eq(%w[Foo/Bar])
+    Example = Struct.new(:context, :comment_cop_names, :all_cop_names, :result)
+    [
+      ['all cops mentioned', 'all', %w[all_names Lint/RedundantCopDisableDirective], %w[all_names]],
+      ['only cop specified', 'Foo/Bar', %w[], %w[Foo/Bar]],
+      ['only department specified', 'Foo', %w[Foo/Bar Foo/Baz Another/Cop], %w[Foo/Bar Foo/Baz]],
+      ['couple departments specified', 'Foo, AnotherFoo',
+       %w[Foo/Bar Foo/Baz AnotherFoo/Bar Another/Cop], %w[Foo/Bar Foo/Baz AnotherFoo/Bar]],
+      ['couple departments and cops specified', 'Foo, AnotherFoo, Baz/Cop',
+       %w[Foo/Bar Foo/Baz AnotherFoo/Bar Another/Cop], %w[Foo/Bar Foo/Baz AnotherFoo/Bar Baz/Cop]],
+      ['Lint department specified', 'Lint',
+       %w[Lint/One Lint/Two Lint/RedundantCopDisableDirective], %w[Lint/One Lint/Two]]
+    ].each do |arr|
```

Comment:
I think using a shared example would be a clearer way to set up these tests. The giant array makes it kind of hard to understand what's going on here IMO.

Reply:
Ow, you've opened new world for me. I didn't know about shared examples. In this case I've rewrote that with separated specs because it required different stubbing.